src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -854,195 +442,13 @@ bool AwtWindow::IsLayered(HWND window)
-    ::KillTimer(GetHWnd(), IDT_AWT_SECURITYANIMATION);
-
-    switch (securityAnimationKind) {
-        case akHide:
-        case akPreHide:
-            ::SetWindowPos(warningWindow, HWND_NOTOPMOST, 0, 0, 0, 0,
-                    SWP_NOACTIVATE | SWP_NOSIZE | SWP_NOMOVE |
-                    SWP_HIDEWINDOW | SWP_NOOWNERZORDER);
-            break;
-        case akShow:
-            RepaintWarningWindow();
-            break;
-    }
-
-    securityAnimationKind = akNone;
-}
-
 MsgRouting AwtWindow::WmTimer(UINT_PTR timerID)
 {
-    if (timerID != IDT_AWT_SECURITYANIMATION) {
-        return mrPassAlong;
-    }
-
-    if (securityWarningAnimationStage == 0) {
-        return mrConsume;
-    }
-
-    securityWarningAnimationStage++;
-    if (securityWarningAnimationStage >= securityAnimationMaxIterations) {
-        if (securityAnimationKind == akPreHide) {
-            // chain real hiding
-            StartSecurityAnimation(akHide);
-        } else {
-            StopSecurityAnimation();
-        }
-    } else {
-        switch (securityAnimationKind) {
-            case akHide:
-                {
-                    BYTE opacity = ((int)0xFF *
-                            (securityAnimationMaxIterations -
-                             securityWarningAnimationStage)) /
-                        securityAnimationMaxIterations;
-                    ::SetLayeredWindowAttributes(warningWindow,
-                            RGB(0, 0, 0), opacity, LWA_ALPHA);
-                }
-                break;
-            case akShow:
-            case akNone: // quite unlikely, but quite safe
-                RepaintWarningWindow();
-                break;
-        }
-    }
-
-    return mrConsume;
-}
-
-// The security warning is visible if:
-//    1. The window has the keyboard window focus, OR
-//    2. The mouse pointer is located within the window bounds,
-//       or within the security warning icon.
-void AwtWindow::UpdateSecurityWarningVisibility()
-{
-    if (!IsUntrusted()) {
-        return;
-    }
-    if (warningWindow == NULL) {
-        return;
-    }
-
-    bool show = false;
-
-    if (IsVisible() && currentWmSizeState != SIZE_MINIMIZED &&
-            !isFullScreenExclusiveMode())
-    {
-        if (AwtComponent::GetFocusedWindow() == GetHWnd()) {
-            show = true;
-        }
-
-        HWND hwnd = AwtToolkit::GetInstance().GetWindowUnderMouse();
-        if (hwnd == GetHWnd()) {
-            show = true;
-        }
-        if (hwnd == warningWindow) {
-            show = true;
-        }
-    }
-
-    if (show && (!::IsWindowVisible(warningWindow) ||
-                securityAnimationKind == akHide ||
-                securityAnimationKind == akPreHide)) {
-        StartSecurityAnimation(akShow);
-    }
-    if (!show && ::IsWindowVisible(warningWindow)) {
-        StartSecurityAnimation(akPreHide);
-    }
+    return mrPassAlong;
 }
 
 void AwtWindow::FocusedWindowChanged(HWND from, HWND to)
```

Comment:
It looks like this empty method and its usage can also be removed:

```
./src/java.desktop/windows/native/libawt/windows/awt_Component.cpp:6426:    AwtWindow::FocusedWindowChanged(old, window);
```

Reply:
I knew that was possible but I wasn't sure I wanted to pull on the string.
If it is really simple I will do it.

Reply:
> I knew that was possible but I wasn't sure I wanted to pull on the string. If it is really simple I will do it.

OK. The string was very short. Fixed.
I believe all comments are now addressed.